pkg/executor/cluster.go

```diff
@@ -102,32 +91,44 @@ func RemoveFinalizer(
 		return nil
 	}
 
-	finalizersPatch := []finalizersPatchSpec{
-		{
-			Op:    "replace",
-			Path:  "/metadata/finalizers",
-			Value: cr.Finalizers,
+	// TODO: Can we just use the cr that was passed in?
+	prevCr := &kdv1.KubeDirectorCluster{}
```

Comment:
More comments about PATCH stuff above.

Reply:
See reply to lengthy PATCH comment above.